File: src/main/java/io/github/binaryfoo/decoders/AsciiPrimitiveDecoder.java
package io.github.binaryfoo.decoders;

import io.github.binaryfoo.tlv.ISOUtil;

public class AsciiPrimitiveDecoder implements PrimitiveDecoder {
    @Override
    public String decode(String hexString) {
        return ISOUtil.dumpString(ISOUtil.hex2byte(hexString));
    }
}

File: src/main/java/io/github/binaryfoo/decoders/AsciiPrimitiveDecoder.kt
package io.github.binaryfoo.decoders

import io.github.binaryfoo.tlv.ISOUtil

class AsciiPrimitiveDecoder : PrimitiveDecoder {
    override fun decode(hexString: String): String {
        return ISOUtil.dumpString(ISOUtil.hex2byte(hexString))
    }
}
